Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1393,12 +1397,43 @@ void LocalFrame::networkUsageDidExceedThreshold()
 
     FRAME_RELEASE_LOG_ERROR(ResourceLoading, "networkUsageDidExceedThreshold: Unloading frame due to exceeding threshold.");
 
-    showResourceMonitoringError("This frame was unloaded."_s);
+    showResourceMonitoringError(generateResourceMonitorErrorHTML());
 
     if (RefPtr document = this->document())
         document->addConsoleMessage(MessageSource::ContentBlocker, MessageLevel::Error, "Frame was unloaded because its network usage exceeded the limit."_s);
 }
 
+String LocalFrame::generateResourceMonitorErrorHTML() const
+{
+#if PLATFORM(COCOA) && HAVE(LOCAL_FRAME_ADDITIONS)
+    return generateResourceMonitorErrorHTMLForCocoa();
+#else
+    return String::fromUTF8(R"(<meta charset="utf-8" />)"
+        "<style>"
+        "body {"
+            "margin:0;padding:0;"
+            "background-color:rgb(242,242,247);"
+            "color:rgb(139,139,143);"
+            "font-family:system-ui,sans-serif; font-size:0.8rem; text-align:center;"
+        "}"
+        "a {"
+            "width:100%; height:100%;"
+            "display:flex; align-items:center; justify-items:center;"
+            "cursor:help;"
+        "}"
+        "span { margin:0 auto; }"
+        "@media (width<60px) { .L,.M { display:none; } }"
+        "@media (60px<=width<160px) { .L,.S { display:none; } }"
+        "@media (width>=160px) { .S,.M { display:none; } }"
+        "</style>"
+        R"(<a title="This frame is hidden for using too many system resources.">)"
+        "<span class=L>⛔︎ Frame hidden for using significant resources</span>"
+        "<span class=M>⛔︎<br />Frame hidden</span>"
+        "<span class=S>⛔︎</span>"
+        "</a>");
```

Comment:
I think we can leave out the fancy HTML for non-Cocoa platforms; if they implement this feature, they can figure out the design that works best for them and change this function as needed.

Another convention is that any string displayed in the UI should go through the WEB_UI_STRING macro, so that our localization scripts (and the localization scripts for ports) know that this is a string that has to be translated to other languages.

So I think I'd just replace this block with something like `return WEB_UI_STRING("This frame is hidden for using too many system resources.", "Heavy frame unloading description")`. You may also need to include `LocalizedStrings.h`.

Reply:
Also, add the localized string key (the first argument to `WEB_UI_STRING` to `Source/WebCore/en.lproj/Localizable.strings`.